test.js

```diff
@@ -112,71 +113,157 @@ test('StuckTransactionsCanceller', async t => {
+          },
+          remove (hash) {
+            assert(storage.has(hash))
+            assert.strictEqual(typeof hash, 'string')
+            storage.delete(hash)
+          }
         },
-        list () {
-          return [...storage.values()]
+        log: str => {
+          // TODO: Test logs
+          debug(str)
         },
-        remove (hash) {
-          assert(storage.has(hash))
-          assert.strictEqual(typeof hash, 'string')
-          storage.delete(hash)
+        sendTransaction (tx) {
+          sentTransactions.push(tx)
+          return {
+            ...tx,
+            hash: 'replacementTxHash',
+            wait: () => {}
+          }
         }
-      },
-      log: str => {
-        // TODO: Test logs
-        debug(str)
-      },
-      sendTransaction (tx) {
-        sentTransactions.push(tx)
-        return {
-          hash: 'replacementTxHash',
-          wait: () => {}
+      })
+      await stuckTransactionsCanceller.addPending(tx)
+      await timers.setImmediate()
+      const status = await stuckTransactionsCanceller.cancelOlderThan(0)
+      assert.deepStrictEqual(status, [{
+        status: 'fulfilled',
+        value: undefined
+      }])
+      assert.strictEqual(sentTransactions.length, 1)
+      const sentTransactionClone = { ...sentTransactions[0] }
+      assert(sentTransactionClone.maxFeePerGas)
+      assert(sentTransactionClone.gasLimit)
+      delete sentTransactionClone.maxFeePerGas
+      delete sentTransactionClone.gasLimit
+      assert.deepStrictEqual(sentTransactionClone, {
+        maxPriorityFeePerGas: 13n,
+        nonce: tx.nonce,
+        from: tx.from,
+        to: tx.from,
+        value: 0
+      })
+      assert.deepStrictEqual(storage, new Map())
+    })
+    await t.test('cancel stuck replacement transactions', async t => {
+      const tx = {
+        hash: 'hash',
+        maxPriorityFeePerGas: 10n,
+        gasLimit: 1n,
+        nonce: 20,
+        from: '0x0'
+      }
+      const storage = new Map()
+      const sentTransactions = []
+      const replacementTxs = [
+        {
+          hash: 'replacementTxHash1',
+          deferred: pDefer()
+        }, {
+          hash: 'replacementTxHash2',
+          deferred: pDefer()
+        }
+      ]
+      let replacementTxIndex = 0
+      const stuckTransactionsCanceller = new StuckTransactionsCanceller({
+        store: {
+          set ({ hash, timestamp, from, maxPriorityFeePerGas, gasLimit, nonce }) {
+            assert(!storage.has(hash))
+            storage.set(hash, {
+              hash,
+              timestamp,
+              from,
+              maxPriorityFeePerGas,
+              gasLimit,
+              nonce
+            })
+          },
+          list () {
+            return [...storage.values()]
+          },
+          remove (hash) {
+            assert(storage.has(hash))
+            assert.strictEqual(typeof hash, 'string')
+            storage.delete(hash)
+          }
+        },
```

Comment:
Is it worth extracting this Store implementation into a shared test helper?

```js
const store = new MemoryStore()
new StuckTransactionsCanceller({
  store,
  // ...
})
// ...

// The replacement transaction is "stuck" now
while (!store.has(replacementTxs[0].hash)) {
  await timers.setTimeout()
}
```

Reply:
Happy to accept follow up PRs 👍

Reply:
It could be an improvement but it's not important enough for me to spend time on right now